From db367cda62d77479085b8154df2728ab536cd5b0 Mon Sep 17 00:00:00 2001
From: Todd Lipcon <todd@cloudera.com>
Date: Tue, 14 Jun 2011 11:02:00 -0700
Subject: [PATCH 01/18] CLOUDERA-BUILD. Add Cloudera-specific build infrastructure.

Major changes:
* Add scripts in cloudera/ directory used for installation from within
  package control scripts
* Add basic hbase.1 man page in cloudera/ dir
* Build should create a "mixed" src/bin tarball, and install from that
* Update pom.xml for CDH:
** Inherit from CDH parent pom and add cdh repo
** Change to CDH version numbering for HBase and dependent projects
** Add cloudera/ directory to assembly
---
 cloudera/README.cloudera  |    9 +++
 cloudera/do-release-build |   22 ++++++++
 cloudera/hbase.1          |   73 ++++++++++++++++++++++++++
 cloudera/install_hbase.sh |  124 +++++++++++++++++++++++++++++++++++++++++++++
 pom.xml                   |   53 ++++++++++++++++----
 src/assembly/all.xml      |    3 +
 6 files changed, 274 insertions(+), 10 deletions(-)
 create mode 100644 cloudera/README.cloudera
 create mode 100755 cloudera/do-release-build
 create mode 100644 cloudera/hbase.1
 create mode 100755 cloudera/install_hbase.sh

diff --git a/cloudera/README.cloudera b/cloudera/README.cloudera
new file mode 100644
index 0000000..8626e8f
--- /dev/null
+++ b/cloudera/README.cloudera
@@ -0,0 +1,9 @@
+This build was generated by Cloudera's build system in the following manner:
+
+1) The pristine open-source release tarball was unpacked
+
+2) The patches contained within the patches/ directory next to this README
+were applied using the apply-patches script. A complete log of these changes
+is also included in CHANGES.cloudera.txt.
+
+3) The project was built by running the do-release-build script in this directory.
diff --git a/cloudera/do-release-build b/cloudera/do-release-build
new file mode 100755
index 0000000..952b536
--- /dev/null
+++ b/cloudera/do-release-build
@@ -0,0 +1,22 @@
+#!/bin/bash
+# Copyright (c) 2009 Cloudera, inc
+#
+# Performs a release build
+
+set -ex
+
+# Do the build
+BIN_DIR=$(readlink -f $(dirname $0))
+RELEASE_DIR=$BIN_DIR/..
+
+cd $RELEASE_DIR
+mvn -DskipTests clean site source:jar install assembly:assembly javadoc:javadoc $DO_MAVEN_DEPLOY
+
+mkdir -p build
+for x in target/*.tar.gz ; do
+  tar -C build -xzf $x
+done
+
+cp -r target/site/apidocs build/hbase-${FULL_VERSION}/docs
+
+(cd build && tar -czf hbase-${FULL_VERSION}.tar.gz hbase-${FULL_VERSION})
diff --git a/cloudera/hbase.1 b/cloudera/hbase.1
new file mode 100644
index 0000000..c5e4a00
--- /dev/null
+++ b/cloudera/hbase.1
@@ -0,0 +1,73 @@
+.\" Process this file with
+.\" groff -man -Tascii hbase.1
+.\"
+.TH hbase 1 "October 2010 " Linux "User Manuals"
+
+.SH NAME
+HBase \- HBase is the Hadoop database.
+
+.SH SYNOPSIS
+
+.B hbase
+\fICOMMAND\fR
+
+.SH DESCRIPTION
+
+HBase is the Hadoop database. Use it when you need random, realtime
+read/write access to your Big Data. This project's goal is the hosting
+of very large tables -- billions of rows X millions of columns -- atop
+clusters of commodity hardware.
+
+HBase is an open-source, distributed, versioned, column-oriented store
+modeled after Google's Bigtable: A Distributed Storage System for
+Structured Data by Chang et al. Just as Bigtable leverages the
+distributed data storage provided by the Google File System, HBase
+provides Bigtable-like capabilities on top of Hadoop.
+
+For more information about HBase, see http://hbase.apache.org.
+
+\fICOMMAND\fR may be one of the following:
+  shell            run the HBase shell
+  shell-tests      run the HBase shell tests
+  zkcli            run the ZooKeeper shell
+  master           run an HBase HMaster node
+  regionserver     run an HBase HRegionServer node
+  zookeeper        run a Zookeeper server
+  rest             run an HBase REST server
+  thrift           run an HBase Thrift server
+  avro             run an HBase Avro server
+  migrate          upgrade an hbase.rootdir
+  hbck             run the hbase 'fsck' tool
+ or
+  CLASSNAME        run the class named CLASSNAME
+
+Most commands print help when invoked w/o parameters or with --help.
+
+.SH ENVIRONMENT
+
+.IP JAVA_HOME
+The java implementation to use.  Overrides JAVA_HOME.
+
+.IP HBASE_CLASSPATH
+Extra Java CLASSPATH entries.
+
+.IP HBASE_HEAPSIZE
+The maximum amount of heap to use, in MB. Default is 1000.
+
+.IP HBASE_OPTS
+Extra Java runtime options.
+
+.IP HBASE_CONF_DIR
+Alternate conf dir. Default is ${HBASE_HOME}/conf.
+
+.IP HBASE_ROOT_LOGGER
+The root appender. Default is INFO,console
+
+.IP HIVE_OPT
+Extra Java runtime options.
+
+.IP HADOOP_HOME
+Optionally, the Hadoop home to run with.
+
+.SH COPYRIGHT
+Copyright (C) 2010 The Apache Software Foundation. All rights reserved.
diff --git a/cloudera/install_hbase.sh b/cloudera/install_hbase.sh
new file mode 100755
index 0000000..bab39d4
--- /dev/null
+++ b/cloudera/install_hbase.sh
@@ -0,0 +1,124 @@
+#!/bin/sh
+# Copyright 2009 Cloudera, inc.
+set -ex
+
+usage() {
+  echo "
+usage: $0 <options>
+  Required not-so-options:
+     --cloudera-source-dir=DIR   path to cloudera distribution files
+     --build-dir=DIR             path to hbase dist.dir
+     --prefix=PREFIX             path to install into
+
+  Optional options:
+     --doc-dir=DIR               path to install docs into [/usr/share/doc/hbase]
+     --lib-dir=DIR               path to install hbase home [/usr/lib/hbase]
+     --installed-lib-dir=DIR     path where lib-dir will end up on target system
+     --bin-dir=DIR               path to install bins [/usr/bin]
+     --examples-dir=DIR          path to install examples [doc-dir/examples]
+     ... [ see source for more similar options ]
+  "
+  exit 1
+}
+
+OPTS=$(getopt \
+  -n $0 \
+  -o '' \
+  -l 'cloudera-source-dir:' \
+  -l 'prefix:' \
+  -l 'doc-dir:' \
+  -l 'lib-dir:' \
+  -l 'installed-lib-dir:' \
+  -l 'bin-dir:' \
+  -l 'examples-dir:' \
+  -l 'build-dir:' -- "$@")
+
+if [ $? != 0 ] ; then
+    usage
+fi
+
+eval set -- "$OPTS"
+while true ; do
+    case "$1" in
+        --cloudera-source-dir)
+        CLOUDERA_SOURCE_DIR=$2 ; shift 2
+        ;;
+        --prefix)
+        PREFIX=$2 ; shift 2
+        ;;
+        --build-dir)
+        BUILD_DIR=$2 ; shift 2
+        ;;
+        --doc-dir)
+        DOC_DIR=$2 ; shift 2
+        ;;
+        --lib-dir)
+        LIB_DIR=$2 ; shift 2
+        ;;
+        --installed-lib-dir)
+        INSTALLED_LIB_DIR=$2 ; shift 2
+        ;;
+        --bin-dir)
+        BIN_DIR=$2 ; shift 2
+        ;;
+        --examples-dir)
+        EXAMPLES_DIR=$2 ; shift 2
+        ;;
+        --)
+        shift ; break
+        ;;
+        *)
+        echo "Unknown option: $1"
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+for var in CLOUDERA_SOURCE_DIR PREFIX BUILD_DIR ; do
+  if [ -z "$(eval "echo \$$var")" ]; then
+    echo Missing param: $var
+    usage
+  fi
+done
+
+MAN_DIR=${MAN_DIR:-/usr/share/man/man1}
+DOC_DIR=${DOC_DIR:-/usr/share/doc/hbase}
+LIB_DIR=${LIB_DIR:-/usr/lib/hbase}
+BIN_DIR=${BIN_DIR:-/usr/lib/hbase/bin}
+ETC_DIR=${ETC_DIR:-/etc/hbase}
+
+install -d -m 0755 $PREFIX/$LIB_DIR
+install -d -m 0755 $PREFIX/$LIB_DIR/lib
+install -d -m 0755 $PREFIX/$DOC_DIR
+install -d -m 0755 $PREFIX/$BIN_DIR
+install -d -m 0755 $PREFIX/$ETC_DIR
+install -d -m 0755 $PREFIX/$MAN_DIR
+
+gzip -c $CLOUDERA_SOURCE_DIR/hbase.1 > $PREFIX/$MAN_DIR/hbase.1.gz
+cp -ra lib/* ${PREFIX}/${LIB_DIR}/lib/
+cp hbase*.jar $PREFIX/$LIB_DIR
+cp -a docs/* $PREFIX/$DOC_DIR
+cp *.txt $PREFIX/$DOC_DIR/
+cp -a hbase-webapps $PREFIX/$LIB_DIR
+
+cp -a conf $PREFIX/$ETC_DIR/conf
+cp -a bin/* $PREFIX/$BIN_DIR
+
+ln -s $ETC_DIR/conf $PREFIX/$LIB_DIR/conf
+
+wrapper=$PREFIX/usr/bin/hbase
+mkdir -p `dirname $wrapper`
+cat > $wrapper <<EOF
+#!/bin/sh
+export ZOOKEEPER_CONF=\${ZOOKEEPER_CONF:-/etc/zookeeper}
+export HADOOP_CONF=\${HADOOP_CONF:-/etc/hadoop-0.20/conf}
+export ZOOKEEPER_HOME=\${ZOOKEEPER_HOME:-/usr/lib/zookeeper}
+export HADOOP_HOME=\${HADOOP_HOME:-/usr/lib/hadoop-0.20}
+export HBASE_CLASSPATH=\$ZOOKEEPER_CONF:\$HADOOP_CONF:\$HADOOP_HOME/*:\$HADOOP_HOME/lib/*:\$ZOOKEEPER_HOME/*:\$ZOOKEEPER_HOME/lib/*:\$HBASE_CLASSPATH
+export HBASE_PID_DIR=/var/run/hbase
+exec /usr/lib/hbase/bin/hbase "\$@"
+EOF
+chmod 755 $wrapper
+
+install -d -m 0755 $PREFIX/usr/bin
diff --git a/pom.xml b/pom.xml
index 3b36890..3e11a67 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1,17 +1,34 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache</groupId>
-    <artifactId>apache</artifactId>
-    <version>8</version>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>cdh-root</artifactId>
+    <version>3.0-u1-SNAPSHOT</version>
   </parent>
 
   <groupId>org.apache.hbase</groupId>
   <artifactId>hbase</artifactId>
   <packaging>jar</packaging>
-  <version>0.90.3</version>
+  <version>0.90.3-cdh3u1-SNAPSHOT</version>
   <name>HBase</name>
   <description>
     HBase is the &amp;lt;a href="http://hadoop.apache.org"&amp;rt;Hadoop&lt;/a&amp;rt; database. Use it when you need
@@ -22,9 +39,8 @@
   <url>http://hbase.apache.org</url>
 
   <scm>
-    <connection>scm:svn:http://svn.apache.org/repos/asf/hbase/tags/0.90.2RC0</connection>
-    <developerConnection>scm:svn:https://svn.apache.org/repos/asf/hbase/tags/0.90.2RC0</developerConnection>
-    <url>http://svn.apache.org/viewvc/hbase/tags/0.90.2RC0</url>
+    <connection>scm:git:git@git.sf.cloudera.com:hbase.git</connection>
+    <url>git@git.sf.cloudera.com:hbase.git</url>
   </scm>
 
   <issueManagement>
@@ -135,6 +151,22 @@
 
   <repositories>
     <repository>
+      <id>cdh.snapshot.repo</id>
+      <url>https://repository.cloudera.com/content/repositories/snapshots</url>
+      <name>Cloudera Snapshot Repository</name>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+    </repository>
+    <repository>
+      <id>cdh.repo</id>
+      <url>https://repository.cloudera.com/content/groups/cloudera-repos</url>
+      <name>Cloudera Repository</name>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+    <repository>
       <id>temp-thrift</id>
       <name>Thrift 0.2.0</name>
       <url>http://people.apache.org/~rawson/repo/</url>
@@ -326,6 +358,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-source-plugin</artifactId>
+        <version>2.1.2</version>
         <executions>
           <execution>
             <id>attach-sources</id>
@@ -452,7 +485,7 @@
     at revision 1034499 with this hdfs-895 patch:
     https://issues.apache.org/jira/secure/attachment/12459473/hdfs-895-branch-20-append.txt
     -->
-    <hadoop.version>0.20-append-r1056497</hadoop.version>
+    <hadoop.version>${cdh.hadoop.version}</hadoop.version>
     <jasper.version>5.5.23</jasper.version>
     <jaxb-api.version>2.1</jaxb-api.version>
     <jetty.version>6.1.26</jetty.version>
@@ -467,7 +500,7 @@
     <slf4j.version>1.5.8</slf4j.version><!-- newer version available -->
     <stax-api.version>1.0.1</stax-api.version>
     <thrift.version>0.2.0</thrift.version><!-- newer version available -->
-    <zookeeper.version>3.3.2</zookeeper.version>
+    <zookeeper.version>${cdh.zookeeper.version}</zookeeper.version>
 
     <!-- For flaky tests exclusion -->
     <test.exclude />
diff --git a/src/assembly/all.xml b/src/assembly/all.xml
index 5ed824a..a072c8a 100644
--- a/src/assembly/all.xml
+++ b/src/assembly/all.xml
@@ -31,6 +31,9 @@
       <fileMode>755</fileMode>
     </fileSet>
     <fileSet>
+      <directory>cloudera</directory>
+    </fileSet>
+    <fileSet>
       <directory>src/main/ruby</directory>
       <outputDirectory>lib/ruby</outputDirectory>
     </fileSet>
-- 
1.7.0.4

